src/perl/lib/wtsi_clarity/epp/isc/analyte_pooler.pm

```diff
@@ -94,16 +94,18 @@ sub _uniq_array {
   my ($self, @array) = @_;
 
   my %seen;
-  return grep { !$seen{$_}++ } @array;
+  my @uniq_array =  grep { !$seen{$_}++ } @array;
+
+  return \@uniq_array;
 }
 
 sub _get_values_from_nodelist {
```

Comment:
This method is being used in a lot of other modules. I'm not sure it belongs in this module. Also, it needs to be public.